script.js

```diff
@@ -38,20 +59,46 @@ const $voiceSelector = document.querySelector('.js-voice-selector');
 const $sayItButton = document.querySelector('.js-say-it-button');
 
 $sayItButton.addEventListener('pointerdown', async () => {
-  await speak();
-  writeDialogueText();
+  await speakAndWriteToDialogueText();
 });
 
+/**
+ * Trims and remove extra white spaces that are between the words of a string.
+ *
+ * @param {string} text The text to be treated.
+ *
+ * @returns {string} The treated string.
+ */
+function trimAndRemoveExtraWhiteSpaces(text) {
```

Comment:
```suggestion
function removeWhiteSpaces(text) {
```

This seems like an appropriate name. Trimming can be considered as a subset of removing white spaces, as it has the same effect.